app/src/test/java/com/android/bookswap/data/source/network/DataConverterTest.kt

```diff
@@ -0,0 +1,80 @@
+package com.android.bookswap.data.source.network
+
+import java.util.UUID
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class DataConverterTest {
+
+  @Test
+  fun parseRawUUID() {
+    val bit_uuid = "{mostSignificantBits=16384, leastSignificantBits=-4611686018427387904}"
+    val str_uuid = "00000000-0000-4000-c000-000000000000"
+    val buuid = DataConverter.parse_raw_UUID(bit_uuid)
+    val suuid = DataConverter.parse_raw_UUID(str_uuid)
+    assertEquals("Converted UUIDs are not equal!", suuid, buuid)
+  }
+
+  @Test
+  fun parseRawUUIDList() {
+    val bit_uuid_list =
+        List(3, { "{mostSignificantBits=16384, leastSignificantBits=${-4611686018427387904+it}}" })
+    val str_uuid_list = List(3, { "00000000-0000-4000-c000-00000000000$it" })
+    val buuids = DataConverter.parse_raw_UUID_list(bit_uuid_list.toString())
+    val suuids = DataConverter.parse_raw_UUID_list(str_uuid_list.toString())
+    assertEquals("Converted UUIDs are not equal!", suuids, buuids)
+  }
+
+  @Test
+  fun parseRawLong() {
+    val raw_long = "1234567890123456789"
+    val big_long: Long = 1234567890123456789
+    val parsed_l = DataConverter.parse_raw_long(raw_long)
+    assertEquals("Value $raw_long was not parsed correctly!!", big_long, parsed_l)
+  }
+
+  @Test
+  fun parseRawLongList() {
+    val big_long: Long = 1234567890123456780
```

Comment:
There are a number of reused variables here, you might want to establish them at the top